core/src/main/scala/cromwell/core/logging/WorkflowLogger.scala

```diff
@@ -0,0 +1,101 @@
+package cromwell.core.logging
+
+import java.nio.file.{Path, Paths}
+
+import akka.actor.{Actor, ActorLogging}
+import akka.event.LoggingAdapter
+import better.files._
+import ch.qos.logback.classic.encoder.PatternLayoutEncoder
+import ch.qos.logback.classic.spi.ILoggingEvent
+import ch.qos.logback.classic.{Level, LoggerContext}
+import ch.qos.logback.core.FileAppender
+import com.typesafe.config.ConfigFactory
+import cromwell.core.WorkflowId
+import lenthall.config.ScalaConfig._
+import org.slf4j.helpers.NOPLogger
+import org.slf4j.{Logger, LoggerFactory}
+
+trait WorkflowLogging extends ActorLogging { this: Actor =>
+  def workflowId: WorkflowId
+
+  lazy val workflowLogger = new WorkflowLogger(self.path.name, workflowId, Option(log))
+}
+
+object WorkflowLogger {
+  def makeFileLogger(path: Path, level: Level): Logger = {
+    val ctx = LoggerFactory.getILoggerFactory.asInstanceOf[LoggerContext]
+    val name = path.getFileName.toString
+
+    Option(ctx.exists(name)) match {
```

Comment:
The appender list is a CopyOnWriteArrayList which provides some thread-safety as well. I will drop a comment here in case we see something break as a hint for future debugging.